File: VB/CustomFunctionCriteriaOperator.Module/BusinessObjects/Company.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace CustomFunctionCriteriaOperator.Module.BusinessObjects
   <DefaultClassOptions()> _
   Public Class Company
       Inherits BaseObject

      Public Sub New(ByVal session As Session)
          MyBase.New(session)
      End Sub


      Private name_Renamed As String
      Public Property Name() As String
         Get
            Return name_Renamed
         End Get
         Set(ByVal value As String)
            SetPropertyValue("Name", name_Renamed, value)
         End Set
      End Property

      <Association("Company-Employees", GetType(Employee))> _
      Public ReadOnly Property Employees() As XPCollection
         Get
            Return GetCollection("Employees")
         End Get
      End Property
   End Class
End Namespace

File: CS/CustomFunctionCriteriaOperator.Module/BusinessObjects/Company.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace CustomFunctionCriteriaOperator.Module.BusinessObjects {
   [DefaultClassOptions()]
   public class Company : BaseObject {

      public Company(Session session)
         : base(session) {}

      private string name;
      public string Name {
         get {
            return name;
         }
         set {
            SetPropertyValue("Name", ref name, value);
         }
      }

      [Association("Company-Employees", typeof(Employee))]
      public XPCollection Employees {
         get {
            return GetCollection("Employees");
         }
      }
   }
}
